### PR TITLE
App footer redesign with certificates

### PR DESCRIPTION
**Pull request by joanpadolina**
@TODO: 
Add the following translation to main Dato env
```
say_hello
EN: Say hello
NL: Neem contact
```
```
visit_us
EN: Visit us
NL: Kom langs
```
## What changes were made
- Add certificate links to DatoCMS
- Add DDA and BCorp svgs
- Redesigned layout
- Remove a ton of styling

## How to test or check results
`scroll down to the footer`

## Preview
Screenshot 2023-06-20 at 13 27 29

## Checks
- [x] All content model changes are done through [scripted migrations](../readme.md#scripted-migrations)
- [x] Appointed PR reviewers

### CODE REVIEW

File: src/components/app-footer/app-footer.vue [hunk 1]
```diff
@@ -43,44 +43,60 @@
...
-              target="_blank"
-              rel="noreferrer noopener"
+
+      <div>
+        <h2 class="h4 app-footer__title">
+          {{ $t('say_hello') }}
+        </h2>
+        <nav
+          aria-label="Contact"
+        >
```
Comment (Frankwarnaar) [inline src/components/app-footer/app-footer.vue:53]: You can wrap the `h2` inside the `nav` element and then remove the `aria-label` from the nav element. Now it's kind of double labeled.

File: src/components/app-footer/app-footer.vue [hunk 2]
```diff
@@ -43,44 +43,60 @@
...
+              >{{ app.emailAddress }}</a>
+            </li>
+          </ul>
+        </nav>
+      </div>
+      <div class="app-footer__list-address-wrapper">
+        <h2 class="h4 app-footer__title">
+          {{ $t('visit_us') }}
+        </h2>
+        <nav
```
Comment (Frankwarnaar) [inline src/components/app-footer/app-footer.vue:76]: Same goes here

**Comment by joanpadolina** [comment]
[Redesign footer with logo's: DDA & B corp](https://trello.com/c/1IQERmB9/633-redesign-footer-with-logos-dda-b-corp)

**Review by Frankwarnaar — CHANGES_REQUESTED** [review]
Fancy change! I like the new titles in the design. Just some semantics and then we're good to go IMO :)

**Review by Siilwyn — CHANGES_REQUESTED** [review]
In Firefox the b-corp logo does not render:
image

Both SVG logos are also not minified it seems? I would pull them through SVGOMG with a lower "number precision" 2/3.
Or even better I think would be to load those images from Dato, then we can use the dato-image component to load optimized images.

File: src/components/app-footer/app-footer.vue (hunk 2 above)
Comment (joanpadolina) [reply src/components/app-footer/app-footer.vue]: ✅

File: migrations/1687269060_addFooterCertificateLinks.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    api_key: "b_corp_link",
+    validators: { format: { predefined_pattern: "url" } },
+    appearance: {
+      addons: [],
+      editor: "single_line",
+      parameters: { heading: false },
+    },
+    default_value: "",
+    fieldset: newFieldsets["712332"],
+  });
```
Comment (Siilwyn) [inline migrations/1687269060_addFooterCertificateLinks.ts:30]: Should we use the `external_link` block for this? @Frankwarnaar

**Review by Siilwyn — CHANGES_REQUESTED** [review]

File: src/components/app-footer/app-footer.vue [hunk 4]
```diff
@@ -102,21 +115,59 @@
...
+              >
+                <span class="sr-only">
+                  {{ certificate.platform }}
+                </span>
+
+                <dato-image
+                  class="app-footer__certificate-logo"
+                  :src="certificate.logo.url"
+                  :width="certificate.logo.width"
+                  :height="certificate.logo.height"
```
[the remark this answers is not in the document]
Comment (joanpadolina) [reply src/components/app-footer/app-footer.vue]: ✅

File: migrations/1687269060_addFooterCertificateLinks.ts (hunk 3 above)
Comment (Frankwarnaar) [reply migrations/1687269060_addFooterCertificateLinks.ts:30]: An external link requires a title as well. I think we could indeed use an external link indeed and then use the title for the screen reader text in the link.

File: migrations/1687269060_addFooterCertificateLinks.ts [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+  console.log(
+    'Create Single-line string field "B corp link" (`b_corp_link`) in model "Footer" (`footer`)'
+  );
+  newFields["9263009"] = await client.fields.create("47595", {
+    label: "B corp link",
+    field_type: "string",
+    api_key: "b_corp_link",
+    validators: { format: { predefined_pattern: "url" } },
```
Comment (Frankwarnaar) [inline migrations/1687269060_addFooterCertificateLinks.ts:22]: Let's make sure to make this field and the other one required.

File: migrations/1687269060_addFooterCertificateLinks.ts [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+  });
+
+  console.log(
+    'Create Single asset field "Bcorp logo" (`bcorp_logo`) in model "Footer" (`footer`)'
+  );
+  newFields["9338640"] = await client.fields.create("47595", {
+    label: "Bcorp logo",
+    field_type: "file",
+    api_key: "bcorp_logo",
+    validators: {
```
Comment (Frankwarnaar) [inline migrations/1687269060_addFooterCertificateLinks.ts:39]: I think the logos should be required (DDA as well)

File: migrations/1687269060_addFooterCertificateLinks.ts (hunk 3 above)
Comment (joanpadolina) [reply migrations/1687269060_addFooterCertificateLinks.ts:30]: Would this be a modular content with a single external link record?
I need to match it with the it logo's, or I can add an image field in the external link and make it not required?

Comment (Siilwyn) [reply migrations/1687269060_addFooterCertificateLinks.ts:30]: > Would this be a modular content with a single external link record?

Yes

>  I need to match it with the it the logo's, or I can add an image field in the external link and make it not required?

Think it's best to 'match it with the logo', as in still have two fields specifically for bcorp & dda, both being a modular content field.

Comment (Frankwarnaar) [reply migrations/1687269060_addFooterCertificateLinks.ts:30]: Could you make sure to make the field labels singular while the api key is plural please? So for instance label the field 'DDA link', while using `dda_links` as api key. Then it looks logical for the content manager, but it also feels logical in the code to pick `ddaLinks[0]`.

Comment (joanpadolina) [reply migrations/1687269060_addFooterCertificateLinks.ts]: ✅

File: migrations/1687269060_addFooterCertificateLinks.ts (hunk 6 above)
Comment (joanpadolina) [reply migrations/1687269060_addFooterCertificateLinks.ts]: ✅

File: migrations/1687269060_addFooterCertificateLinks.ts (hunk 5 above)
Comment (joanpadolina) [reply migrations/1687269060_addFooterCertificateLinks.ts]: ✅

**Review by Siilwyn — CHANGES_REQUESTED** [review]
Don't forget to remove the logo svg assets

File: src/components/app-footer/app-footer.vue [hunk 7]
```diff
@@ -102,21 +115,55 @@
             </a>
           </li>
         </ul>
+
+        <ul class="app-footer__certificate-list">
+          <template
+            v-for="certificate in certificateLinks"
+            :key="certificate.url"
+          >
+            <li v-if="certificate.url !== ''">
```
[the remark this answers is not in the document]
Comment (joanpadolina) [reply src/components/app-footer/app-footer.vue]: True, the whole template can be removed since we dont need to check anymore. Thanks!

**Review by Siilwyn — APPROVED** [review]
Yes let's go!

File: migrations/1687269060_addFooterCertificateLinks.ts (hunk 3 above)
Comment (joanpadolina) [reply migrations/1687269060_addFooterCertificateLinks.ts]: @Frankwarnaar Do we actually want the shortened version of dda or was that an example (kinda agree it's getting long).

Comment (Frankwarnaar) [reply migrations/1687269060_addFooterCertificateLinks.ts]: It was an example. I'm not sure

**Review by Frankwarnaar — APPROVED** [review]

**Pull request merged**